packages/task/src/browser/task-definition-registry.ts

```diff
@@ -15,14 +15,25 @@
  ********************************************************************************/
 
 import { injectable } from 'inversify';
+import { Disposable, DisposableCollection, Event, Emitter } from '@theia/core/lib/common';
 import { TaskConfiguration, TaskCustomization, TaskDefinition } from '../common';
 
 @injectable()
-export class TaskDefinitionRegistry {
+export class TaskDefinitionRegistry implements Disposable {
 
     // task type - array of task definitions
     private definitions: Map<string, TaskDefinition[]> = new Map();
 
+    protected readonly onDidChangeTaskDefinitionEmitter = new Emitter<void>();
+    get onDidChangeTaskDefinition(): Event<void> {
```

Comment:
`onDidChangeTaskDefinition` is confusing, does not mean that an existing task defintion is changed?

How about `onDidChange` (meaning registry state is change) or `onDidRegisterTaskDefinition` (meaning is obvious)